File: Irrgarten_Ruby/Shield.rb
#encoding:utf-8

require_relative 'Dice'

module Irrgarten 
	class Shield
		
		def initialize (protection, uses) #float, int
			@protection = protection
			@uses = uses	
		end
		
		def protect ()
			salida = 0.0
			if @uses > 0 then 
				salida = @protection
				@uses -= 1
			end
			salida
		end
		
		def to_s ()
			"S[#{@protection},#{@uses}]" 
		end
		
		def discard 
			return Dice.discard_element(@uses)
		end
	end	
end

File: Irrgarten_Java/Irrgarten_P4/src/irrgarten/Shield.java
package irrgarten;

public class Shield extends CombatElement{
    
    public Shield (float protection, int uses) {
        super(protection,uses);
    }
    
    public float protect() {
        return super.produceEffect();
    }
    
    @Override
    public String toString () { 
        String temp = super.toString();
        return "S" + temp;
    }
}